dot/network/transaction.go

```diff
@@ -119,6 +119,40 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
+func (s *Service) createBatchMessageHandler(txnBatch chan *batchMessage) NotificationsMessageBatchHandler {
+	txnBatchHandler := func(peer peer.ID, msg NotificationsMessage) (msgs []*batchMessage, err error) {
```

Comment:
could we return the function directly?

Reply:
Updated the code to return function. 
We cannot make the channel just receiver as we are also reading data out from it inside the callback.